cachito/workers/pkg_managers/yarn.py

```diff
@@ -69,7 +70,33 @@ def get_yarn_proxy_repo_username(request_id):
     return f"cachito-yarn-{request_id}"
 
 
-def _get_deps(yarn_lock, file_deps_allowlist):
+def _find_reachable_deps(yarn_lock, dep):
+    """
+    Get a set of all dependencies reachable by a top-level dependency using BFS.
+
+    :param yarn_lock: yarn.lock file as a dictionary
+    :param dep: top-level dependency in package.json
+    :return: set of all dependencies reachable by dep
+    :rtype: set
+    """
+    yarn_lock_parsed = _expand_replacements(yarn_lock)
+    bfs_queue = deque([dep])
+    reachable_deps = set([dep])
```

Comment:
This looks like py2.6 way
```suggestion
    reachable_deps = {dep, }
```